File: leetcode/editor/en/[557]Reverse Words in a String III.java
557
        Reverse Words in a String III
        2023-01-23 20:57:27

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public String reverseWords(String s) {
        StringBuilder sb = new StringBuilder();
        s = s.trim();
        String[] words = s.split("\s+");
        for (String word : words) {
            char[] ch = word.toCharArray();
            for (int i = ch.length - 1; i >= 0; i--) {
                sb.append(ch[i]);
            }
            sb.append(" ");
        }
        return sb.toString().trim();
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: leetcode/editor/en/[557]Reverse Words in a String III.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function (s) {
    return s.split(' ').map(word => word.split('').reverse().join('')).join(' ');
};


//leetcode submit region end(Prohibit modification and deletion)
/*
var reverseWords = function (s) {
    const c = s.split('');
    let left = 0;
    let right = 0;

    while (right < c.length) {
        while (right < c.length && c[right] !== ' ') {
            right++;
        }

        reverseWord(left, right - 1, c);
        left = right + 1;
        right = left;
    }

    return c.join('');
};

function reverseWord(left, right, c) {
    while (left < right) {
        [c[left], c[right]] = [c[right], c[left]];
        left++;
        right--;
    }
}
 */